src/uthash.h

```diff
@@ -105,6 +102,30 @@ typedef unsigned char uint8_t;
 #define uthash_expand_fyi(tbl)            /* can be defined to log expands   */
 #endif
 
+#ifndef HASH_NOMEM_OK
+#define HASH_NOMEM_OK 0
+#endif
+
+#if HASH_NOMEM_OK
+/* malloc failures can be recovered from */
+
+#ifndef uthash_fatal
+#define uthash_fatal(obj) do{}while(0)
```

Comment:
Previously I had them separated into `uthash_fatal` and `uthash_mem_failed`, and you have requested that they should be joined together because they are both called in case of OOM, and are nearly identical (and poorly named, yes, but that should be accepted). You are now requesting that they are to be separated. Can you please confirm?

The `uthash_oom` is to be called for every OOM (I'm talking for `HASH_NOMEM_OK=1` case only here). Granted, it is not called at the exact time of the OOM, but only after we released any memory that has been allocated, and restored all state to before the hash call was attempted;  all that is because we allow the caller to jump out of the current stack (using `throw`/`longjmp`, etc.). If an OOM has occurred during the add process, no more memory allocation attempts are performed, so you also can't have multiple malloc failures per call, making it called once per OOM. Even in case of `HASH_SELECT`, it's called once per OOM.